src/components/ObjectHelper.js

```diff
@@ -0,0 +1,87 @@
+import TypeErrorException from './BaseExceptions';
+
+class ObjectHelper {
+
+	static objectSize( Obj ) {
+		let Key;
+		let Size = 0;
+		for ( Key in Obj ) {
+			if ( Obj.hasOwnProperty( Key ) === true ) {
+				Size++;
+			}
+		}
+		return Size;
+	}
+
+	static isEmpty( Obj ) {
+		return ObjectHelper.objectSize( Obj ) === 0;
+	}
+
+	static copyObj( Obj, Depth = -1 ) {
+		let Dolly, Key;
+
+		if ( typeof Obj !== 'object' ) {
+			throw new TypeErrorException(
+				`Expected object got ${ typeof arguments[ 0 ] } at argument 0.`
+			);
+		}
+
+		if ( Array.isArray( Obj ) === true ) {
+			Dolly = Obj.slice( 0 );
+		} else {
+			Dolly = Object.assign( {}, Obj );
+		}
+
+		if ( Depth === 0 ) {
+			return Dolly;
+		}
+
+		for ( Key in Dolly ) {
+			if ( typeof Dolly[ Key ] === 'object' ) {
+				Dolly[ Key ] = ObjectHelper.copyObj( Dolly[ Key ], Depth - 1 );
+			}
+		}
+
+		return Dolly;
+	}
+
+	static mergeObj() {
```

Comment:
see the comment about array/object spreads

Reply:
See the comment about arrays/object spreads